seccomp_test.go

```diff
@@ -5,11 +5,63 @@
 package seccomp
 
 import (
+	"bytes"
 	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
 	"syscall"
 	"testing"
+	"time"
+	"unsafe"
 )
 
+// execInSubprocess calls the go test binary again for the same test.
+// This must be only top-level statment in the test function. Do not nest this.
+// It will slightly defect the test log output as the test is entered twice
+func execInSubprocess(t *testing.T, f func(t *testing.T)) {
+	const subprocessEnvKey = `GO_SUBPROCESS_KEY`
+	if testIDString, ok := os.LookupEnv(subprocessEnvKey); ok && testIDString == "1" {
+		t.Run(`subprocess`, f)
+		return
+	}
+
+	cmd := exec.Command(os.Args[0])
+	cmd.Args = []string{os.Args[0], "-test.run=" + t.Name() + "$", "-test.v=true"}
+	for _, arg := range os.Args {
+		if strings.HasPrefix(arg, `-test.testlogfile=`) {
+			cmd.Args = append(cmd.Args, arg)
+		}
+	}
+	cmd.Env = []string{subprocessEnvKey + "=1"}
+	cmd.Stdin = os.Stdin
+
+	var b bytes.Buffer
```

Comment:
using `strings.Builder` instead of `bytes.Buffer` would not require the new import and not the `string(b.Bytes())` cast.

Reply:
Updated.